custom/enikshay/tasks.py

```diff
@@ -163,10 +167,13 @@ def run_batch(self, case_ids):
             updates = []
             for episode in self._get_open_episode_cases(case_ids):
                 did_error = False
+                dose_status_by_date = None
                 update_json = {}
                 for updater in self.updaters:
                     try:
                         potential_update = updater(self.domain, episode).update_json()
+                        if isinstance(updater, EpisodeAdherenceUpdate):
```

Comment:
You should put this into the `EpisodeAdherenceUpdate` class instead.

Reply:
you can do something like 
```
ledger_updates += getattr(updater(self.domain, episode), ledger_updates)
``` 
which will then allow you to batch them in the same way as we do for case updates. (and then rename the `update_json` in all the updaters to `case_update_json` or something more descriptive. ).